lib/api/apiUtils/object/coldStorage.js

```diff
@@ -34,6 +34,42 @@ function getAmzRestoreResHeader(objMD) {
     return undefined;
 }
 
+/**
+ * set archiveInfo headers to target header object
+ * @param {object} objMD - object metadata
+ * @returns {object} headers - target header object
+ */
+function setArchiveInfoHeaders(objMD) {
+    const headers = {};
+
+    if (objMD['x-amz-scal-transition-in-progress']) {
+        headers['x-amz-scal-transition-in-progress'] = true;
+        headers['x-amz-scal-transition-time'] = new Date(objMD['x-amz-scal-transition-time']).toUTCString();
+    }
+
+    if (objMD.archive) {
+        headers['x-amz-scal-archive-info'] = JSON.stringify(objMD.archive.archiveInfo);
+
+        if (objMD.archive.restoreRequestedAt) {
+            headers['x-amz-scal-restore-requested-at'] = new Date(objMD.archive.restoreRequestedAt).toUTCString();
+            headers['x-amz-scal-restore-requested-days'] = objMD.archive.restoreRequestedDays;
+        }
+
+        if (objMD.archive.restoreCompletedAt) {
+            headers['x-amz-scal-restore-completed-at'] = new Date(objMD.archive.restoreCompletedAt).toUTCString();
+            headers['x-amz-scal-restore-will-expire-at'] = new Date(objMD.archive.restoreWillExpireAt).toUTCString();
+        }
+    }
+
+    // Always get the "real" storage class (even when STANDARD) in this case
+    headers['x-amz-storage-class'] = objMD['x-amz-storage-class'] || objMD.dataStoreName;
```

Comment:
I think this won't work as `x-amz-storage-class` is never empty. It's set to `STANDARD` when the object is hot. But in any case, using `x-amz-storage-class` all the time should work too as it's set to the cold location name when the object is transitioned and is kept that way even on restores.

Reply:
I think - it won't hurt keeping a fallback in case of any issue (or different API path leading to an empty x-amz-storage-class somehow?)

Reply:
yes i agree